sdk/identity/identity/src/credentials/interactiveBrowserCredential.ts

```diff
@@ -6,21 +6,64 @@
 import { TokenCredential, GetTokenOptions, AccessToken } from "@azure/core-http";
 import { InteractiveBrowserCredentialOptions } from "./interactiveBrowserCredentialOptions";
 import { credentialLogger, formatError } from "../util/logging";
+import { TokenCredentialOptions, IdentityClient } from "../client/identityClient";
+import { DefaultTenantId, DeveloperSignOnClientId } from "../constants";
+import { Socket } from "net";
+
+const SERVER_PORT = process.env.PORT || 80;
+
+import express from "express";
+import msal from "@azure/msal-node";
+import open from "open";
+import path from "path";
+import http from "http";
 
 const BrowserNotSupportedError = new Error(
   "InteractiveBrowserCredential is not supported in Node.js."
 );
 const logger = credentialLogger("InteractiveBrowserCredential");
 
+interface AuthenticationRecord {
+  authority?: string,
+  homeAccountId: string,
+  environment: string,
+  tenantId: string,
+  username: string,
+}
+
 /**
  * Enables authentication to Azure Active Directory inside of the web browser
  * using the interactive login flow, either via browser redirects or a popup
  * window.  This credential is not currently supported in Node.js.
  */
 export class InteractiveBrowserCredential implements TokenCredential {
+  private identityClient: IdentityClient;
+  private pca: msal.PublicClientApplication;
+  private msalCacheManager: msal.TokenCache;
+  private tenantId: string;
+  private clientId: string;
+  private persistenceEnabled: boolean;
+  private authenticationRecord: AuthenticationRecord | undefined;
+
   constructor(options?: InteractiveBrowserCredentialOptions) {
-    logger.info(formatError(BrowserNotSupportedError));
-    throw BrowserNotSupportedError;
+    this.identityClient = new IdentityClient(options);
+    this.tenantId = (options && options.tenantId) || DefaultTenantId;
+    this.clientId = (options && options.clientId) || DeveloperSignOnClientId;
+    
+    // Future update: this is for persistent caching
+    this.persistenceEnabled = false;
+    this.authenticationRecord = undefined;
+
+    const publicClientConfig = {
+      auth: {
+        clientId: this.clientId,
+        authority: "https://login.microsoftonline.com/" + this.tenantId,
+        redirectUri: "http://localhost"
```

Comment:
We should make `redirectUrl` by adding an option to `InteractiveBrowserCredentialOptions`. We have updated other languages to make this configurable for this preview, so it would be great it JS did this right out of the gate.

Reply:
Good catch, I'll add it. So we'll take it in as `redirectUrl` and then point `redirectUri` to it?

Reply:
Yes, it should still default to "http://localhost" if it's not specified in the `options`.

Reply:
Looks like we already take in a `redirectUri` from the browser version. We could reuse that.

Reply:
roger that. updated.